sql/src/main/java/io/crate/protocols/postgres/ConnectionContext.java

```diff
@@ -344,8 +357,50 @@ public void channelDisconnected(ChannelHandlerContext ctx, ChannelStateEvent e)
 
     private void handleStartupBody(ChannelBuffer buffer, Channel channel) {
         session = readStartupMessage(buffer);
-        Messages.sendAuthenticationOK(channel);
+        if (authService != null && authService.enabled()) {
+            authenticate(channel);
+        } else {
+            Messages.sendAuthenticationOK(channel);
+            sendReadyForQuery(channel);
+        }
+    }
+
+    private void authenticate(Channel channel) {
+        SessionContext ctx = session.sessionContext();
+        InetAddress address = ((InetSocketAddress) channel.getRemoteAddress()).getAddress();
```

Comment:
yes, because the channel is always `NioSocketChannel` which returns InetSocketAddress

```java
    @Override
    InetSocketAddress getRemoteSocketAddress() throws Exception {
        return (InetSocketAddress) channel.socket().getRemoteSocketAddress();
    }
```

maybe it's better to cast the channel and then call `getRemoteAddress()`?